templates/includes/pagination.html

```diff
@@ -0,0 +1,15 @@
+<ul class="pagination">
+  {% if page_obj.has_previous %}
+    <li class="page-item">
+      <a href="?page= {{ page_obj.previous_page_number }}" class="page-link">prev</a>
```

Comment:
There is an extra space in the href attribute of the anchor tags for the previous and next page links. It should be `href="?page={{ page_obj.previous_page_number }}"` and `href="?page={{ page_obj.next_page_number }}"` without the space after `page=`.